src/layer/tile/GridLayer.js

```diff
@@ -232,7 +232,7 @@ export class GridLayer extends Layer {
 	redraw() {
 		if (this._map) {
 			this._removeAllTiles();
-			const tileZoom = this._clampZoom(this._map.getZoom());
+			const tileZoom = this._clampZoom(Math.round(this._map.getZoom()));
```

Comment:
Just thinking out loud. Instead of rounding the number, perhaps we should ceil it? That way, the resolution of the tile will always be high enough, and not get blurry. Then again, you might want to get a lower zoom level tile, not quite sure.